src/Couchbase.Lite.Shared/API/DI/Service.cs

```diff
@@ -61,12 +62,9 @@ public static class Service
         /// <exception cref="ArgumentNullException">Thrown if <paramref name="assembly"/> is <c>null</c></exception>
         /// <exception cref="InvalidOperationException">Thrown if an invalid type is found inside of the assembly (i.e.
         /// one that does not implement any interfaces and/or does not have a parameter-less constructor)</exception>
-        [ContractAnnotation("null => halt")]
-        public static void AutoRegister(Assembly assembly)
+        public static void AutoRegister([NotNull]Assembly assembly)
         {
-            if (assembly == null) {
-                throw new ArgumentNullException(nameof(assembly));
-            }
+            CBDebug.MustNotBeNull(WriteLog.To.Database, Tag, nameof(assembly), assembly);
```

Comment:
`CBDebug` cannot be used inside of this method because it will try to log stuff, which will access the DI stuff that hasn't been registered yet (it will be registered later in this method).  This will cause a dependency injection not found exception.

Reply:
I see, should I use Debug.Assert instead?

Reply:
No, just leave it the way it was before since this is a user-callable method.